backend/de.metas.contracts/src/main/java/de/metas/contracts/commission/salesrep/process/CreateMissingCommissionEntriesForSalesRep.java

```diff
@@ -0,0 +1,428 @@
+import ch.qos.logback.classic.Level;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import de.metas.bpartner.BPartnerId;
+import de.metas.bpartner.service.BPartnerQuery;
+import de.metas.bpartner.service.IBPartnerBL;
+import de.metas.bpartner.service.IBPartnerDAO;
+import de.metas.contracts.commission.commissioninstance.interceptor.C_Invoice_CandidateFacadeService;
+import de.metas.invoicecandidate.api.IInvoiceCandDAO;
+import de.metas.invoicecandidate.api.InvoiceCandidateMultiQuery;
+import de.metas.invoicecandidate.api.InvoiceCandidateQuery;
+import de.metas.invoicecandidate.model.I_C_Invoice_Candidate;
+import de.metas.logging.LogManager;
+import de.metas.process.JavaProcess;
+import de.metas.user.UserId;
+import de.metas.user.api.IUserDAO;
+import de.metas.util.Loggables;
+import de.metas.util.NumberUtils;
+import de.metas.util.Services;
+import de.metas.util.time.InstantInterval;
+import de.metas.util.time.IntervalUtils;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+import org.adempiere.ad.column.AdColumnId;
+import org.adempiere.ad.table.ChangeLogEntry;
+import org.adempiere.ad.table.ChangeLogEntryQuery;
+import org.adempiere.ad.table.ChangeLogEntryRepository;
+import org.adempiere.ad.table.api.AdTableId;
+import org.adempiere.ad.table.api.IADTableDAO;
+import org.compiere.SpringContextHolder;
+import org.compiere.model.I_AD_User;
+import org.compiere.model.I_C_BPartner;
+import org.compiere.util.TimeUtil;
+import org.slf4j.Logger;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class CreateMissingCommissionEntriesForSalesRep extends JavaProcess
+{
+	private static final Logger logger = LogManager.getLogger(CreateMissingCommissionEntriesForSalesRep.class);
+
+	private final ChangeLogEntryRepository changeLogEntryRepository = SpringContextHolder.instance.getBean(ChangeLogEntryRepository.class);
+	private final C_Invoice_CandidateFacadeService invoiceCandFacadeService = SpringContextHolder.instance.getBean(C_Invoice_CandidateFacadeService.class);
+
+	private final IADTableDAO adTableDAO = Services.get(IADTableDAO.class);
+	private final IBPartnerDAO bPartnerDAO = Services.get(IBPartnerDAO.class);
+	private final IUserDAO userDAO = Services.get(IUserDAO.class);
+	private final IBPartnerBL bPartnerBL = Services.get(IBPartnerBL.class);
+	private final IInvoiceCandDAO invoiceCandDAO = Services.get(IInvoiceCandDAO.class);
+
+	private static final String FROM_DATE = "2020-01-01";
+	private static final String TO_DATE = "2020-12-31";
+
+	@Override
+	protected String doIt() throws Exception
```

Comment:
If think it makes sense to annotate the `doIt()` method with `@RunOutOfTrx`. that way the doIt() method is not run in a trx that might eventually time out and be killed